app/models/manageiq/providers/redhat/inventory/collector/target_collection.rb

```diff
@@ -0,0 +1,164 @@
+class ManageIQ::Providers::Redhat::Inventory::Collector::TargetCollection < ManageIQ::Providers::Redhat::Inventory::Collector
+  def initialize(_manager, _target)
+    super
+    parse_targets!
+    infer_related_ems_refs!
+
+    # Reset the target cache, so we can access new targets inside
+    target.manager_refs_by_association_reset
+  end
+
+  def clusters
+    if target.kind_of? VmOrTemplate
```

Comment:
right, seems like you still count with only 1 target being here.

If should be that the target is a TargetCollection, then target.targets could contain VmOrTemplate object